File: java/demos/ps/maps/Drasgow.java
package demos.ps.maps;

import static demos.ps.oo.PSGame.*;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import demos.ps.oo.*;
import demos.ps.oo.PSGame.Planet;
import demos.ps.oo.PSMenu.EntityClothes;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.Scene;

public class Drasgow {

	private static final Logger log = LogManager.getLogger(Drasgow.class);

	public static void house1() {
		PSMenu.startScene(Scene.BLUE_HOUSE, EntityType.CITY_MAN_BLOND, EntityClothes.RED);
		PSMenu.Stext(getString("Drasgow_House_1"));
		PSMenu.endScene();
	}

	public static void house2() {
		PSMenu.startScene(Scene.YELLOW_HOUSE, EntityType.CITY_WMN_BROWN, EntityClothes.BLUE);
		PSMenu.Stext(getString("Drasgow_House_2"));
		PSMenu.endScene();
	}
	
	public static void ent1() {
		EntStart();
		PSMenu.Stext(getString("Drasgow_People_Ent1"));
		EntFinish();
	}
	public static void ent2() {
		EntStart();
		PSMenu.Stext(getString("Drasgow_People_Ent2"));
		EntFinish();
	}
	public static void ent3() {
		EntStart();
		PSMenu.Stext(getString("Drasgow_People_Ent3"));
		EntFinish();
	}	

	public static void exit() {
		mapswitch(Planet.PALMA, City.DRASGOW.getX(), City.DRASGOW.getY());		
	}
	
	public static void tunnel() {
		mapswitch(Dungeon.DRASGOW_DUNGEON);
	}

	
	
}

File: src/demos/ps/maps/Drasgow.ts
/**
 * Drasgow City Script
 * TypeScript port of Drasgow.java
 */

import { PSGame } from '../PSGame';
import { Planet, City, CityHelper } from '../game/City';
import { Dungeon } from '../game/Dungeon';
import { PSSceneType, EntityType, EntityClothes, PSMenu } from '../PSMenu';

export class Drasgow {

  public static async house1(): Promise<void> {
    await PSMenu.startScene(PSSceneType.BLUE_HOUSE, EntityType.CITY_MAN_BLOND, EntityClothes.RED);
    await PSMenu.Stext(PSGame.getString("Drasgow_House_1"));
    await PSMenu.endScene();
  }

  public static async house2(): Promise<void> {
    await PSMenu.startScene(PSSceneType.YELLOW_HOUSE, EntityType.CITY_WMN_BROWN, EntityClothes.BLUE);
    await PSMenu.Stext(PSGame.getString("Drasgow_House_2"));
    await PSMenu.endScene();
  }

  public static async ent1(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Drasgow_People_Ent1"));
    PSGame.EntFinish();
  }

  public static async ent2(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Drasgow_People_Ent2"));
    PSGame.EntFinish();
  }

  public static async ent3(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Drasgow_People_Ent3"));
    PSGame.EntFinish();
  }

  public static exit(): void {
    PSGame.mapswitchToPlanet(Planet.PALMA, CityHelper.getX(City.DRASGOW), CityHelper.getY(City.DRASGOW));
  }

  public static async tunnel(): Promise<void> {
    await PSGame.mapswitchToDungeon(Dungeon.DRASGOW_DUNGEON);
  }
}
